### PR TITLE
Remove the resizeDetector and use document-container for the event hand…

### PR DESCRIPTION
**Pull request by gokaysatir**
…ling.

Change-Id: I6990480da8ba3a058e4e5d32407bf38d341e8ee4

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Review by eszkadev — CHANGES_REQUESTED** [review]
it's green week, let's wait for red. I'm blocking it for 1 day, so it is marked as reviewed

**Review by mmeeks — COMMENTED** [review]
Would be lovely to see this go =)

**Comment by eszkadev** [comment]
If you make window small, then open sidebar, then resize to bigger window. Sidebar stays small:
sidebar

File: browser/src/docevents.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        // Resize event for a div element is not fired. We'll use a resize-observer for catching the event.
+        new ResizeObserver(this.fire.bind(this, 'resize')).observe(this.container);
+    }
+
+    public fire(eventType: string, details: any) {
+        const newEvent = new CustomEvent(eventType, details);
+        this.container.dispatchEvent(newEvent);
+    }
+
+    public on(eventType: string, callback: eventCallback) {
```
Comment (eszkadev) [inline browser/src/docevents.ts:41]: we need also `off` I think

File: browser/.beforeprettier [hunk 2]
```diff
@@ -92,6 +92,7 @@
 /src/core/Socket.js
 /src/core/Util.js
 /src/docstate.js
+/src/docevents.ts
```
Comment (eszkadev) [inline browser/.beforeprettier:95]: We can have later more and more "global app helpers".
Can we have directory for them maybe?
I think "app" would be good, it fits into the name we use for accessing them so can be quickly found by newcomers.
Also in the whole app we have convention to use CamelCase naming for files. Only in the main directory are some exceptions. I would prefer to keep all unified: `DocEvents.ts`

so: `app/DocEvents.ts` would be nice to begin the trend (later at new Big version we could move also docstate and docstatefunctions)

Comment (eszkadev) [reply browser/.beforeprettier:95]: and please not include that file here, it is meant to keep only OLD files which we don't want to format with prettier

File: browser/src/docevents.ts (hunk 1 above)
Comment (gokaysatir) [reply browser/src/docevents.ts:41]: I couldn't decide on this. Not sure if we have a real need for off functionality here. Let's keep that out for now, if we see a scenario that needs, then we can always add it quickly.

File: browser/.beforeprettier (hunk 2 above)
Comment (gokaysatir) [reply browser/.beforeprettier:95]: And thanks for checking:)

**Review by eszkadev — APPROVED** [review]

**Pull request merged**